sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/rows.scala

```diff
@@ -37,6 +35,21 @@ trait MutableRow extends Row {
   def setByte(ordinal: Int, value: Byte)
   def setFloat(ordinal: Int, value: Float)
   def setString(ordinal: Int, value: String)
+
+  final def setByte(ordinal: Int, value: Int) { setByte(ordinal, value.asInstanceOf[Byte]) }
+  final def setShort(ordinal: Int, value: Int) { setByte(ordinal, value.asInstanceOf[Short]) }
+
+  final def update(bound: BoundReference, value: Any) { update(bound.ordinal, value) }
+  final def setInt(bound: BoundReference, value: Int) { setInt(bound.ordinal, value) }
+  final def setLong(bound: BoundReference, value: Long) { setLong(bound.ordinal, value) }
+  final def setDouble(bound: BoundReference, value: Double) { setDouble(bound.ordinal, value) }
+  final def setBoolean(bound: BoundReference, value: Boolean) { setBoolean(bound.ordinal, value) }
+  final def setShort(bound: BoundReference, value: Int) { setShort(bound.ordinal, value) }
+  final def setShort(bound: BoundReference, value: Short) { setShort(bound.ordinal, value) }
+  final def setByte(bound: BoundReference, value: Byte) { setByte(bound.ordinal, value) }
+  final def setByte(bound: BoundReference, value: Int) { setByte(bound.ordinal, value) }
+  final def setFloat(bound: BoundReference, value: Float) { setFloat(bound.ordinal, value) }
```

Comment:
There is the same issue of the `Row` interface changes.
Plz see the comment of `Row`.